### ISSUE TITLE
Getting 500 Internal Server Error when fetching "userinfo" with a "service account" token

### ISSUE THREAD

**Issue by Nizar-Rahme** [keycloak/keycloak-community#270]
**Edit**
This only happens in version 12.* but works fine in version 11.*
----
In the Keycloak server, we created a client that has an `Access Type` of `confidential`.
By calling `/protocol/openid-connect/token` endpoint with the `client_id` and the `client_secret`, we got the `access_token`, which was supposed to authenticate the application (client) in question in the follow-up requests. 

Even though the token is valid (which I double-checked by calling the `introspect` endpoint), I am getting a 500 Error from the server for any request that has this `access_token`

So in short:
- Calling this endpoint `<base_url>/auth/realms/<realm>/protocol/openid-connect/token` gives an access_token
  - Provided data: client_id,client_secret, grant_type: "client_credentials"
- Calling `<base_url>/auth/realms/<realm>/protocol/openid-connect/userinfo` gives `500Internal Server Error` 

```JSON
{
    "error": "unknown_error"
}
```

Any insights on what the issue could be?

**Comment by stianst** [comment]
This is not the correct place to report bugs or ask for help. Please see https://www.keycloak.org/community for more information.
